parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -1183,6 +1183,8 @@ LogicalTypeAnnotation getLogicalTypeAnnotation(LogicalType type) {
         return LogicalTypeAnnotation.uuidType();
       case FLOAT16:
         return LogicalTypeAnnotation.float16Type();
+      case VARIANT:
+        return LogicalTypeAnnotation.variantType();
```

Comment:
This needs to handle `specification_version`.

Reply:
@rdblue, @emkornfield I have some basic questions on this specification_version. We already the version (that should be this specification_version?) in the variant header.  

I can see that we define a constant of `VARIANT_SPECIFICATION_VERSION` and the writer will write that into the header. Then the reader should be able to fetch the version and parse the variant if it's supported.

Variant logical type may not  be associated with a version. Let me know if I misunderstand anything.

Reply:
Synced offline. The spec specifies the supported latest version while the data can be of different versions. 

With that, variant logical type will be like: `variant(1)` with 1 to be the version.

Reply:
Sorry for the delay, this sounds right to me.